src/tp_list_maker_server.cpp

```diff
@@ -0,0 +1,377 @@
+
+  boolean prev_check = false;
+  int miniverse_index = 0; 
+  //TODO(shaneallcroft): (after launchpad presentations) accomodate a room system
+  for (size_t x = 0; x < map.info.width; ++x) {
+    for (size_t y = 0; y < map.info.height; ++y) {
+      /* I _think_ this is right -- I'll check the GMapping source. */
+      size_t idx = map.info.width * x + y;
+      if(map.data[idx] == 0 && x_min == -2){  /* weird if it finds an isolated open space it cant get to but i dont think thats very plausible*/
+	x_min = x;
+	y_min = y;
+      }
+      else if(map.data[idx] == 0){
+	x_max = x;  //no other requirements, this is because the x and y will always be at their highest
+	y_max = y;
+      }
+    }
+  }
+  
+  res.miniverse[] = new hide_and_seek::tp_msg[ (x_max - x_min) * (y_max - y_min) ];  //tp miniverse
+  // ^^ is this legal? i dont see why it would be
+  for(int x = x_min; x <= x_max; ++x)
+  {
+    for(int y = y_min; y <= y_max; ++y)
+    {
+      /*
+	the following code iterates thru every spot on the grid in the room to see if it is occupied
+	if it is it the 8 surrounding spots on the grid are analyzed, if they are free, a tourist point is marked at the central spot first checked at the beginning of the loop
+	it could be done with a single if statement with a ton of && operators but thats so fucking ugly and this way shorts circuits anyway soooo.......
+*/
+      
+      size_t idx = map.info.width * x + y;/* Thank you Hunter! :) */
+      if(prev_check)
+      {
+	if(map.data[idx - map.info.width + 1] == 0)
+	{
+	  if(map.data[idx + 1] == 0)
+	  {
+	    if(map.data[idx + map.info.width + 1] == 0)
+	    {
+	        ROS_INFO("Tourist point marked at ( %ld , %ld )",(long int) x, (long int) y);
+		hide_and_seek::tp_msg tourist_point;
+		tourist_point.tp_pos.x = x;
+		tourist_point.tp_pos.y = y;
+		tourist_point.visited = req.miniverse[miniverse_index].visited;
+		res.miniverse[miniverse_index] = tourist_point;
+		++miniverse_index;
+		prev_check = true;
+		continue;
+	    }
+	  }
+	}        
+      }
+      if(map.data[idx] == 0)
+      {
+	if(map.data[idx - map.info.width - 1] == 0)
+	{
+	  if(map.data[idx - map.info.width] == 0)
+	  {
+	    if(map.data[idx - map.info.width + 1] == 0)
+	    {
+	      if(map.data[idx - 1] == 0)
+	      {
+		if(map.data[idx + 1] == 0)
+		{
+		  if(map.data[idx + map.info.width - 1] == 0)
+		  {
+		    if(map.data[idx + map.info.width] == 0)
+		    {
+		      if(map.data[idx + map.info.width + 1] == 0)
+		      {
+			 ROS_INFO("Tourist point marked at ( %ld , %ld )",(long int) x, (long int) y);
+			 hide_and_seek::tp_msg tourist_point;
+			 tourist_point.tp_pos.x = x;
+			 tourist_point.tp_pos.y = y;
+			 tourist_point.visited = req.miniverse[miniverse_index].visited;
+			 //tourist_point.postcard = false;
+			 res.miniverse[miniverse_index] = tourist_point;
+			 ++miniverse_index;
+			 prev_check = true;
+			 continue;
+		      }
+		    }
+		  }
+		}
+	      }
+	    }
+	  }
+	}  
+      }
```

Comment:
What in the hell?


----


But actually though, I have a tip:

```
if (condition) {
  if (other_condition) {
    if (yet_another_condition) {
      /* do stuff */
    }
  }
}
```

could also be written like this:

```
if (condition && other_condition && yet_another_condition) {
  /* do stuff */
}
```

And this is in fact better than doing a for loop to evaluate by the way. That's what most scrubs would do. Good on you.